.github/workflows/lint.yml

```diff
@@ -0,0 +1,19 @@
+name: Linters
```

Comment:
do we want clippy in here?

Reply:
we currently do not check clippy in CI and have a number of warnings. this would fail right now so i think this is worth tackling separately.

Reply:
got it! do we have an issue?

Reply:
#373 sorta